### PR TITLE
Refactor tinfoil hat snowflake code

### PR DESCRIPTION
**Pull request by adacovsk**
[system]

With tinfoil hat, it was common to see `iswearing(tinfoil_hat, head)...` all throughout the code. There are two other items just like it, the `stun helmet` and `martian bubblehelm` so you have to check for all three. There were more under tracking.dm for AI. There is also a check called `tinfoil_check`. Then there's `digitalcamo`, which already existed. There's also `M_PSY_RESIST` that was used. All of this should be checked inside `.can_mind_interact`.

## What this does
- Replaces snowflake code into a simple variable
- Fixes some weird code/optimizes
- Makes use of `.can_mind_interact` instead of checking M_PSY_RESIST flag
- removes `digitalcamo` variable

## Why it's good
- Eliminates long lines of conditions
- Easier to change things or add things in the future (which I plan on doing).
- removes lots of unnecessary checks

This was a nightmare to fix, so there's one less nightmare

🆑 
 * tweak: Changed some abilities and items to be more consistent related to tracking, viewing, and telepathy. For example, Changeling's Digital Camouflage is now like a tinfoil hat. Certain syndicate items now jam various technology and mind powers like a tinfoil hat.

### CODE REVIEW

**Comment by Kurfursten** [comment]
Why would you do this as a var that updates on equipped instead of just making it a proc that checks current conditions? This would be much less flexible and no risk of somehow breaking it (for example, melting off the headwear without unequipping it, just as a thought).

**Comment by ghost** [comment]
> Why would you do this as a var that updates on equipped instead of just making it a proc that checks current conditions? This would be much less flexible and no risk of somehow breaking it (for example, melting off the headwear without unequipping it, just as a thought).

Funny you say that because I just coded clothing to call unequip procs when they get melted off, as a way to set their vars on people's slots to null while deleting.

**Comment by Kurfursten** [comment]
It's great that you thought of that particular thing too, but this implementation is too dangerous because it counts on you thinking of every possible bug. A proc is more OOP, flexible, and safer.

**Comment by Exxion** [comment]
Any means of removing something from a slot should unequip it, and if it doesn't, that's a bug. However, there are still benefits of having it be a check rather than a var

**Comment by adacovsk** [comment]
The variable already exists for lings so that's why I thought to use it. I can use a proc though

**Comment by DamianX** [comment]
>Changelogs should represent how a player might be affected by the changes, not a summary of the PR's contents. Therefore, if something only affects administrators, you can leave it out.

**Comment by Armadingus** [comment]
But I don't want to teleport with a tinfoil hat, how am I supposed to survive DJKramer's wormhole events?

**Comment by DamianX** [comment]
I don't get it. Why would a syndicate agent ID or the changeling's digital camouflage prevent me from talking telepathically to them? Why would a voice changer give immunity to mind-affecting spells?

**Comment by ManopolisJones** [comment]
Hey, this is all great and all, but I like to not be teleported with a tinfoil hat on.

**Comment by adacovsk** [comment]
Okay, I make the tinfoil hat prevent teleporting again.

**Comment by adacovsk** [comment]
> I don't get it. Why would a syndicate agent ID or the changeling's digital camouflage prevent me from talking telepathically to them? Why would a voice changer give immunity to mind-affecting spells?

I wanted to give these items a little more function. What they currently do is prevent AI tracking, like they possess some jamming tech. I can add it to their descriptions.

The more I think about it, telepathy shouldn't be affected

**Comment by Kurfursten** [comment]
Tinfoil hat as counterplay to people getting grabbed by telescience was always the intention by the way (an alternative to shutting off suit sensors), so I'm glad that's being kept

File: code/game/dna/genes/goon_powers.dm [hunk 1]
```diff
@@ -12,16 +12,16 @@
 	block = SOBERBLOCK
 
 //WAS: /datum/bioEffect/psychic_resist
-/datum/dna/gene/basic/psychic_resist
+/datum/dna/gene/basic/interference
 	name = "Psy-Resist"
```
Comment (Kurfursten) [inline code/game/dna/genes/goon_powers.dm]: Isn't it weird you didn't change the name here even though you did on the injector?

Comment (adacovsk) [reply code/game/dna/genes/goon_powers.dm]: yes

**Comment by adacovsk** [comment]
> Tinfoil hat as counterplay to people getting grabbed by telescience was always the intention by the way (an alternative to shutting off suit sensors), so I'm glad that's being kept

I wasn't sure to make it so that all of them can prevent it or just keep it as the hat... I'd rather call a proc than have one specific item do it

**Comment by gurfan** [comment]
Hey, this is all great and all, but I like to not be mindraped by grey's and mushrooms with a tinfoil hat on.

**Comment by adacovsk** [comment]
>Hey, this is all great and all, but I like to not be mindraped by grey's and mushrooms with a tinfoil hat on.

You serious?

**Comment by gurfan** [comment]
Why else would I wear a tinfoil hat.

**Comment by Kurfursten** [comment]
He poses a valid point in that getting annoying "voice in your head" messages really only has one reasonable counter.

**Comment by adacovsk** [comment]
Okay gurfan. You got what you want

**Comment by Kurfursten** [comment]
It occurs to me that digital camo now will make you really obvious to anyone wearing a sechud or medhud. "Why is that hatless guy not on my hud? Ling!"

Probably want to update wiki for that

**Pull request merged**